examples/golang/components/http-password-checker/main.go

```diff
@@ -0,0 +1,79 @@
+//go:generate go run github.com/bytecodealliance/wasm-tools-go/cmd/wit-bindgen-go generate --world hello --out gen ./wit
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	gopasswordvalidator "github.com/wagslane/go-password-validator"
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+type CheckRequest struct {
+	Value string `json:"value"`
+}
+
+type CheckResponse struct {
+	Valid  bool `json:"valid"`
+	Length int  `json:"length"`
+}
+
+func init() {
+	// Register the handleRequest function as the handler for all incoming requests.
+	wasihttp.HandleFunc(handleRequest)
+}
+
+func checkHandler(w http.ResponseWriter, r *http.Request) {
+	var req CheckRequest
+	defer r.Body.Close()
+
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		errResponseJSON(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	if err := json.Unmarshal(b, &req); err != nil {
+		errResponseJSON(w, http.StatusBadRequest, fmt.Sprintf("error with json input: %s", err.Error()))
+		return
+	}
+
+	err = gopasswordvalidator.Validate(req.Value, 60)
+	if err != nil {
+		errResponseJSON(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	resp := CheckResponse{Valid: true, Length: len(req.Value)}
+	respJSON, err := json.Marshal(resp)
+	if err != nil {
+		errResponseJSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(respJSON)
+}
```

Comment:
A slightly cleaner alternative might read something like this:
```suggestion
	w.Header().Set("Content-Type", "application/json")
	err = json.NewEncoder(w).Encode(resp)
	if err != nil {
		errResponseJSON(w, http.StatusInternalServerError, err.Error())
	}
}
```